pkg/integration/helpers.go

```diff
@@ -0,0 +1,343 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package integration contains verification server integration tests.
+package integration
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/sha1"
+	"fmt"
+	"math/big"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/exposure-notifications-server/pkg/keys"
+	"github.com/google/exposure-notifications-server/pkg/server"
+	"github.com/google/exposure-notifications-verification-server/pkg/cache"
+	"github.com/google/exposure-notifications-verification-server/pkg/config"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller/certapi"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller/codestatus"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller/issueapi"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller/middleware"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller/verifyapi"
+	"github.com/google/exposure-notifications-verification-server/pkg/database"
+	"github.com/google/exposure-notifications-verification-server/pkg/ratelimit"
+	"github.com/google/exposure-notifications-verification-server/pkg/render"
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+	"github.com/mikehelmick/go-chaff"
+)
+
+const (
+	realmName       = "test-realm"
+	realmRegionCode = "test"
+	adminKeyName    = "integration-admin-key"
+	deviceKeyName   = "integration-device-key"
+)
+
+// Suite contains the integration test configs and other useful data.
+type Suite struct {
+	cfg *config.IntegrationTestConfig
+
+	db    *database.Database
+	realm *database.Realm
+
+	adminKey, deviceKey string
+
+	adminSrv *server.Server
+	apiSrv   *server.Server
+}
+
+// NewTestSuite creates a Suite for integration tests.
+func NewTestSuite(tb testing.TB, ctx context.Context) *Suite {
+	tb.Helper()
+	cfg, db := config.NewIntegrationTestConfig(ctx, tb)
+
+	// Create or reuse the existing realm
+	realm, err := db.FindRealmByName(realmName)
+	if err != nil {
+		if !database.IsNotFound(err) {
+			tb.Errorf("error when finding the realm %q: %w", realmName, err)
+		}
+		realm = database.NewRealmWithDefaults(realmName)
+		realm.RegionCode = realmRegionCode
+		if err := db.SaveRealm(realm); err != nil {
+			tb.Errorf("failed to create realm %+v: %w: %v", realm, err, realm.ErrorMessages())
+		}
+	}
+
+	// Create new API keys
+	suffix, err := randomString()
+	if err != nil {
+		tb.Errorf("failed to create suffix string for API keys: %w", err)
+	}
+
+	adminKey, err := realm.CreateAuthorizedApp(db, &database.AuthorizedApp{
+		Name:       adminKeyName + suffix,
+		APIKeyType: database.APIUserTypeAdmin,
+	})
+	if err != nil {
+		tb.Errorf("error trying to create a new Admin API Key: %w", err)
```

Comment:
I think all of these should be fatal. In general, use fatal when you don't want execution to continue.